### PR TITLE
SLING-8523 : Duplicate OSGI configs after upgrade

### PR DESCRIPTION
**Pull request by ashokpanghal**
Upgrade resulting in duplicate factory configs for cases where alais is null

### CODE REVIEW

**Comment by ashokpanghal** [comment]
@cziegeler , please review

File: src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java [hunk 1]
```diff
@@ -77,27 +77,57 @@ public void update(final Collection<UpdatableResourceGroup> groups) {
...
+            factoryPid = alias.substring(0, pos - 1);
+            pid = oldId.substring(factoryPid.length() + 1);
+        } else {
+            // extract factory id for these cases where alias is not available and factoryId and pid need to be separated from the old id string itself
+            //format assumption ::: "factory_pid.factory_pid.pid"
+            // split pid with lastIndexOf('.') then remove the duplicate factory_pid part from the remaining string using the middle dot split index
 
-        final String factoryPid = alias.substring(0, pos - 1);
-        final String pid = oldId.substring(factoryPid.length() + 1);
+            String factoryIdString = oldId.substring(0,oldId.lastIndexOf('.')+1); // keep it +1 to have last dot intact so that we always have even dots in the string
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java]: you could store lastIndexOf in an int variable and reuse it below

File: src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java [hunk 2]
```diff
@@ -77,27 +77,57 @@ public void update(final Collection<UpdatableResourceGroup> groups) {
...
+            factoryPid = oldId.substring(0,getMiddleDotSplitIndex(factoryIdString,'.'));
+            pid = oldId.substring(oldId.lastIndexOf('.')+1,oldId.length());
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUpdateHandler.java]: substring(int) is enough, if you only specify the start it goes until the of the string

**Comment by ashokpanghal** [comment]
@cziegeler , Thanks for the review, added a test case to cover the new code and incorporated your code comments. Please merge. 
On a side note, sonar code analysis is still showing a failure despite 100% code coverage for newly added code

**Pull request merged**